test/e2e/framework/deployments.go

```diff
@@ -0,0 +1,228 @@
+
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DeploymentType int
+
+const (
+	InvalidDeploymentPodType DeploymentType = iota
+	NginxDeployment
+	NetShootDeployment
+)
+
+type DeploymentConfig struct {
+	Type         DeploymentType
+	Cluster      ClusterIndex
+	ServiceIP    string
+	ReplicaCount int32
+	PodName      string
+	// TODO: namespace, once https://github.com/submariner-io/submariner/pull/141 is merged
+}
+type Deployment struct {
+	PodList            *v1.PodList
+	Deployment         *appsv1.Deployment
+	Config             *DeploymentConfig
+	TerminationCode    int32
+	TerminationMessage string
+	framework          *Framework
+}
+
+func (f *Framework) NewDeployment(config *DeploymentConfig) *Deployment {
+
+	// check if all necessary details are provided
+	Expect(config.Type).ShouldNot(Equal(InvalidDeploymentPodType))
+
+	deployment := &Deployment{Config: config, framework: f, TerminationCode: -1}
+
+	switch config.Type {
+	case NetShootDeployment:
+		deployment.buildNetshootDeployment()
+	case NginxDeployment:
+		deployment.buildNginxDeployment()
+	}
+
+	return deployment
+}
+
+func (d *Deployment) AwaitReady() {
+	pods := d.framework.ClusterClients[d.Config.Cluster].CoreV1().Pods(d.framework.Namespace)
+	d.PodList, _ = pods.List(metav1.ListOptions{})
+	d.PodList = AwaitUntil("get pod", func() (interface{}, error) {
+		podList, err := pods.List(metav1.ListOptions{})
+		if nil != err {
+			Logf("Error while retrieving podlist")
+			return nil, err
+		}
+		d.PodList = podList
+		return podList, nil
+	}, func(result interface{}) (bool, error) {
+		if d.PodList == nil || len(d.PodList.Items) == 0 {
+			return false, nil
+		}
+		for _, pod := range d.PodList.Items {
+			if pod.Status.Phase != v1.PodRunning {
+				if pod.Status.Phase != v1.PodPending {
+					return false, nil
+				}
+				return false, nil // pod is still pending
+			}
+		}
+		return true, nil // pod is running
+	}).(*v1.PodList)
+}
+
+func (d *Deployment) AwaitSuccessfulFinish() {
+	pods := d.framework.ClusterClients[d.Config.Cluster].CoreV1().Pods(d.framework.Namespace)
+
+	d.PodList = AwaitUntil("get pod", func() (interface{}, error) {
+		podList, err := pods.List(metav1.ListOptions{})
+		if nil != err {
+			Logf("Error while retrieving podlist")
+			return nil, err
+		}
+		d.PodList = podList
+		return podList, nil
+	}, func(result interface{}) (bool, error) {
+		if d.PodList.Items == nil {
+			Logf("no pods present")
+			return false, nil
+		}
+		if d.PodList == nil || len(d.PodList.Items) == 0 {
+			return false, nil
+		}
+		for _, pod := range d.PodList.Items {
+			Logf("pod.Status.Phase %v", pod.Status.Phase)
+			switch pod.Status.Phase {
+			case v1.PodRunning:
+				Logf("Continue")
+				continue
+			default:
+				Logf("Returning false %v", pod.Status.Phase)
+				return false, nil
+			}
+		}
+		Logf("Return default true")
+		return true, nil
```

Comment:
The code above returns true if all pods are running so this function actually does the opposite as it's name implies and effectively does the same thing as `AwaitReady`.  This function can be removed.

Reply:
Done.